### PR TITLE
Wip scrub misc

### PR DESCRIPTION
**Pull request by dzafman**
For Kraken from previous work:
    Scrub fixes for semantics of snapshot results.  Add test of snapset corruption.
    Minor cleanup and logging improvements.

### CODE REVIEW

**Comment by dzafman** [comment]
Jenkins only 1 unrelated failure

  79/146 Test  #86: unittest_erasure_code_plugin_jerasure ...***Exception: SegFault  0.22 sec

File: src/osd/ReplicatedBackend.cc [hunk 1]
```diff
@@ -789,7 +789,8 @@ void ReplicatedBackend::be_deep_scrub(
   ScrubMap::object &o,
   ThreadPool::TPHandle &handle)
 {
-  dout(10) << __func__ << " " << poid << " seed " << seed << dendl;
+  dout(10) << __func__ << " " << poid << " seed "
```
Comment (tchaikov) [inline src/osd/ReplicatedBackend.cc]: no need to restore the iostream state by sending `std::dec` to it. `dout` creates a new `ceph::log::Entry` instance. for every `dout()`. the `ostream` under the hood is the `ceph::log::Entry::m_streambuf` variable, and they are not shared.

Comment (dzafman) [inline src/osd/ReplicatedBackend.cc]: I'm going to leave it so it is there should anything be added to the end of that message.

**Comment by dzafman** [comment]
jenkins test this please

**Pull request merged**